### PR TITLE
OSDOCS#4551: Add the common terms for networking book

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.8+

Issue: https://issues.redhat.com/browse/OSDOCS-4551

Link to docs preview: https://52972--docspreview.netlify.app/openshift-enterprise/latest/networking/understanding-networking.html#nw-networking-glossary-terms_understanding-networking

QE review:
- [x] QE has approved this change.

Additional information:

This content has QE acks from this [PR](https://github.com/openshift/openshift-docs/pull/52765) which got automatically closed.

### CODE REVIEW

**Comment by sheriff-rh** [comment]
Please add the build preview to the top:
https://52972--docspreview.netlify.app/openshift-enterprise/latest/networking/understanding-networking.html#nw-networking-glossary-terms_understanding-networking

**Review by sheriff-rh — CHANGES_REQUESTED** [review]
Overall this content is very good and was much needed. 

Many of my comments are proofreading-related, such as sentence capitalization or missing periods. Please also remember to include a build preview in the description. 

To improve this content at a later time, consider moving it to it's own assembly. You can `xref::` to  modules in the documentation where customers can learn more about a topic. 

It's possible some of these terms may not be available in all distributions, which could cause confusion for some customers. 

In all - this is a much needed improvement to our docs, thank you!

File: modules/nw-networking-glossary-terms.adoc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+route::
+The {product-title} route provides Ingress traffic to services in the cluster. Routes provide advanced features that might not be supported by standard Kubernetes Ingress Controllers, such as TLS re-encryption, TLS passthrough, and split traffic for blue-green deployments.
+
+scaling::
+Increasing or decreasing the resource capacity.
+
+service::
+Exposes a running application on a set of pods.
+
+Single Root I/O Virtualization (SR-IOV) Network Operator:: The Single Root I/O Virtualization (SR-IOV) Network Operator manages the SR-IOV network devices and network attachments in your cluster.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+{product-title} Ingress Operator::
+The Ingress Operator implements the `IngressController` API and is the component responsible for enabling external access to {product-title} services.
+
+pod::
+One or more containers with shared resources, such as volume and IP addresses, running in your {product-title} cluster.
+A pod is the smallest compute unit defined, deployed, and managed.
+
+PTP Operator::
+The PTP Operator creates and manages the linuxptp services.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+A primary node agent that runs on each node in the cluster to ensure that containers are running in a pod.
+
+Kubernetes NMState Operator::
+The Kubernetes NMState Operator provides a Kubernetes API for performing state-driven network configuration across the {product-title} cluster’s nodes with NMState.
+
+kube-proxy::
+Kube-proxy is a proxy service which runs on each node and helps in making services available to the external host. It helps in forwarding the request to correct containers and is capable of performing primitive load balancing.
+
+Load balancers::
+{product-title} uses load balancer for communicating from outside the cluster with services running in the cluster.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+Cluster DNS is a DNS server which serves DNS records for Kubernetes services. Containers started by Kubernetes automatically include this DNS server in their DNS searches.
+
+DNS Operator::
+The DNS Operator deploys and manages CoreDNS to provide a name resolution service to pods. This enables DNS-based Kubernetes Service discovery in {product-title}.
+
+deployment::
+A Kubernetes resource object that maintains the life cycle of an application.
+
+domain::
+domain is a DNS name serviced by the Ingress Controller.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+This glossary defines common terms that are used in the architecture content. These terms help you understand {product-title} architecture effectively
+
+authentication::
+To control access to an {product-title} cluster, a cluster administrator can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.
+
+AWS Load Balancer Operator::
+The AWS Load Balancer (ALB) Operator deploys and manages an instance of the aws-load-balancer-controller.
+
+Certificates::
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+[id="nw-networking-glossary-terms_{context}"]
+= Common terms
+
+This glossary defines common terms that are used in the architecture content. These terms help you understand {product-title} architecture effectively
+
+authentication::
+To control access to an {product-title} cluster, a cluster administrator can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.
+
+AWS Load Balancer Operator::
+The AWS Load Balancer (ALB) Operator deploys and manages an instance of the aws-load-balancer-controller.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 7]
```diff
@@ -0,0 +1,119 @@
+// Module included in the following assemblies:
+//
+// * networking/understanding-networking.adoc
+
+:_content-type: REFERENCE
+[id="nw-networking-glossary-terms_{context}"]
+= Common terms
+
+This glossary defines common terms that are used in the architecture content. These terms help you understand {product-title} architecture effectively
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 8]
```diff
@@ -0,0 +1,119 @@
+// Module included in the following assemblies:
+//
+// * networking/understanding-networking.adoc
+
+:_content-type: REFERENCE
+[id="nw-networking-glossary-terms_{context}"]
+= Common terms
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+kubelet::
+A primary node agent that runs on each node in the cluster to ensure that containers are running in a pod.
+
+Kubernetes NMState Operator::
+The Kubernetes NMState Operator provides a Kubernetes API for performing state-driven network configuration across the {product-title} cluster’s nodes with NMState.
+
+kube-proxy::
+Kube-proxy is a proxy service which runs on each node and helps in making services available to the external host. It helps in forwarding the request to correct containers and is capable of performing primitive load balancing.
+
+Load balancers::
```
Comment (bergerhoffer) [inline modules/nw-networking-glossary-terms.adoc:65]: L should be lowercase here

**Review by bergerhoffer — COMMENTED** [review]
Just a few minor things if you wouldn't mind taking a look before I merge?

File: modules/nw-networking-glossary-terms.adoc [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+As a cluster administrator, you can add the MetalLB Operator to your cluster so that when a service of type LoadBalancer is added to the cluster, MetalLB can add an external IP address for the service.
+
+multicast::
+With IP multicast, data is broadcast to many IP addresses simultaneously.
+
+namespaces::
+A namespace isolates specific system resources that are visible to all processes. Inside a namespace, only processes that are members of that namespace can see those resources.
+
+networking::
+Network information of {product-title} cluster.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+The Kubernetes NMState Operator provides a Kubernetes API for performing state-driven network configuration across the {product-title} cluster’s nodes with NMState.
+
+kube-proxy::
+Kube-proxy is a proxy service which runs on each node and helps in making services available to the external host. It helps in forwarding the request to correct containers and is capable of performing primitive load balancing.
+
+Load balancers::
+{product-title} uses load balancers for communicating from outside the cluster with services running in the cluster.
+
+MetalLB Operator::
+As a cluster administrator, you can add the MetalLB Operator to your cluster so that when a service of type LoadBalancer is added to the cluster, MetalLB can add an external IP address for the service.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

File: modules/nw-networking-glossary-terms.adoc (hunk 9 above)
Comment (xenolinux) [reply modules/nw-networking-glossary-terms.adoc]: acked

**Comment by xenolinux** [comment]
> Just a few minor things if you wouldn't mind taking a look before I merge?

@bergerhoffer Thanks for the review! I addressed the changes

**Comment by bergerhoffer** [comment]
Thanks, LGTM now! Merging

**Pull request merged**